### PR TITLE
Add documentation for generic deployment

### PR DESCRIPTION
**Pull request by pyromaniackeca**

### CODE REVIEW

File: source/docs/generic-deployment.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+## SSH key
+
+The next step is optional, if your server requires an ssh key for deployment
+you need to copy the contents of your private key here.
+
+<img src ="/docs/assets/img/generic-deployment/ssh-key.png" class="img-responsive img-bordered">
+
+We recommend generating one specifically for Semaphore. 
+We will safely encrypt it. Make sure that the corresponding public key is
```
Comment (strika) [inline source/docs/generic-deployment.md]: Perhaps "Semaphore will safely encrypt it".

File: source/docs/generic-deployment.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title: Generic Deployment
+category: Deploying your application
+---
+
+To get started go to your project's page and click on the "Set Up Deployment"
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: To get started, go (missing comma)

File: source/docs/generic-deployment.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+button.
+
+<img src ="/docs/assets/img/generic-deployment/project-page.png" class="img-responsive img-bordered">
+
+This will launch a simple five step process.
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: five-step

File: source/docs/generic-deployment.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title: Generic Deployment
+category: Deploying your application
+---
+
+To get started go to your project's page and click on the "Set Up Deployment"
+button.
+
+<img src ="/docs/assets/img/generic-deployment/project-page.png" class="img-responsive img-bordered">
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: Alt tags missing from all images.

File: source/docs/generic-deployment.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+## Deployment strategy
+
+The next step is choosing your deployment strategy.
+Semaphore supports two deployment strategies, Automatic and Manual.
+You can read more about them [here](/docs/understanding-automatic-and-manual-deployment.md).
+
+<img src ="/docs/assets/img/generic-deployment/deployment-strategy.png" class="img-responsive img-bordered">
+
+If you chose automatic deployment, you need to chose a branch whose successfull
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: successful

File: source/docs/generic-deployment.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+## Deployment strategy
+
+The next step is choosing your deployment strategy.
+Semaphore supports two deployment strategies, Automatic and Manual.
+You can read more about them [here](/docs/understanding-automatic-and-manual-deployment.md).
+
+<img src ="/docs/assets/img/generic-deployment/deployment-strategy.png" class="img-responsive img-bordered">
+
+If you chose automatic deployment, you need to chose a branch whose successfull 
+build we want to trigger a deploy.
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: we want to -> should

File: source/docs/generic-deployment.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+The next step is choosing your deployment strategy.
+Semaphore supports two deployment strategies, Automatic and Manual.
+You can read more about them [here](/docs/understanding-automatic-and-manual-deployment.md).
+
+<img src ="/docs/assets/img/generic-deployment/deployment-strategy.png" class="img-responsive img-bordered">
+
+If you chose automatic deployment, you need to chose a branch whose successfull 
+build we want to trigger a deploy. 
+If you chose manual deployment this step is skipped.
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: manual deployment, this (missing comma)

File: source/docs/generic-deployment.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+## Deploy commands
+
+Enter custom commands you need Semaphore to execute to deploy your 
+application.
+
+<img src ="/docs/assets/img/generic-deployment/deploy-commands.png" class="img-responsive img-bordered">
+
+## SSH key
+
+The next step is optional, if your server requires an ssh key for deployment
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: This step is optional. If your server requires an SSH key for deployment,

File: source/docs/generic-deployment.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+you need to copy the contents of your private key here.
+
+<img src ="/docs/assets/img/generic-deployment/ssh-key.png" class="img-responsive img-bordered">
+
+We recommend generating one specifically for Semaphore. 
+We will safely encrypt it. Make sure that the corresponding public key is 
+present in server's authorized_keys list.
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: in the server's authorized_keys list

File: source/docs/generic-deployment.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+The next step is optional, if your server requires an ssh key for deployment
+you need to copy the contents of your private key here.
+
+<img src ="/docs/assets/img/generic-deployment/ssh-key.png" class="img-responsive img-bordered">
+
+We recommend generating one specifically for Semaphore. 
+We will safely encrypt it. Make sure that the corresponding public key is 
+present in server's authorized_keys list. 
+
+Through these steps you give Semaphore permission to access your server and
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: Through these steps, you (missing comma)

File: source/docs/generic-deployment.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+We recommend generating one specifically for Semaphore. 
+We will safely encrypt it. Make sure that the corresponding public key is 
+present in server's authorized_keys list. 
+
+Through these steps you give Semaphore permission to access your server and 
+successfully execute deploy commands.
+
+**Note:** The private key will be encrypted on Semaphore and will be available
+only during the deployment process. After the initial setup there is no way to
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: After the initial setup, there (missing comma)

File: source/docs/generic-deployment.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+change it through Semaphore interface. You can identify it with the fingerprint
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: through the Semaphore interface

File: source/docs/generic-deployment.md (hunk 11 above)
Comment (dunjardl) [inline source/docs/generic-deployment.md]: there is no way to -> it is not possible to

File: source/docs/generic-deployment.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+on the project settings page.
+
+## Server name
+
+Choose a name for the Server, Semaphore will generate a name
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: Next, you can choose a name for the Server. Semaphore will automatically generate

File: source/docs/generic-deployment.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+for you, but it is recommended to change it to something meaningfull for your
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: meaningful for (maybe change it to "related to")

File: source/docs/generic-deployment.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title: Generic Deployment
+category: Deploying your application
+---
+
+Semaphore has built in support for Heroku, Capistrano and Cloud66. However if
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: built-in support
However, if

File: source/docs/generic-deployment.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+To get started, go to your project's page and click on the "Set Up Deployment"
+button.
+
+<img alt="project-page" src="/docs/assets/img/generic-deployment/project-page.png" class="img-responsive img-bordered">
+
+This will launch a simple five-step process.
+
+## Deployment method
+
+The first step is to choose generic deployment.
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: choose -> select

File: source/docs/generic-deployment.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+
+The next step is choosing your deployment strategy.
+Semaphore supports two deployment strategies, Automatic and Manual.
+You can read more about them [here](/docs/understanding-automatic-and-manual-deployment.md).
+
+<img alt="deployment-strategy" src="/docs/assets/img/generic-deployment/deployment-strategy.png" class="img-responsive img-bordered">
+
+If you chose automatic deployment, you need to chose a branch whose successfull 
+build we want to trigger a deploy. 
+If you chose manual deployment this step is skipped.
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: manual deployment, this step

File: source/docs/generic-deployment.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+
+## Deployment strategy
+
+The next step is choosing your deployment strategy.
+Semaphore supports two deployment strategies, Automatic and Manual.
+You can read more about them [here](/docs/understanding-automatic-and-manual-deployment.md).
+
+<img alt="deployment-strategy" src="/docs/assets/img/generic-deployment/deployment-strategy.png" class="img-responsive img-bordered">
+
+If you chose automatic deployment, you need to chose a branch whose successfull
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: successful

File: source/docs/generic-deployment.md [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+To get started, go to your project's page and click on the "Set Up Deployment"
+button.
+
+<img src="/docs/assets/img/generic-deployment/project-page.png" alt="project-page" class="img-responsive img-bordered">
+
+This will launch a simple five-step process.
+
+## Deployment method
+
+The first step is to select generic deployment.
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: Generic Deployment

File: source/docs/generic-deployment.md [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+## Deploy commands
+
+Enter custom commands you need Semaphore to execute to deploy your 
+application.
+
+<img src="/docs/assets/img/generic-deployment/deploy-commands.png" alt="deploy-commands" class="img-responsive img-bordered">
+
+## SSH key
+
+This step is optional. If your server requires an ssh key for deployment
```
Comment (dunjardl) [inline source/docs/generic-deployment.md]: SSH
deployment,

**Pull request merged**